recipes/xerces-c/all/conanfile.py

```diff
@@ -29,6 +29,8 @@ def _build_subfolder(self):
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
+        else:
+            del self.options.char_type
```

Comment:
Why removing the option if the default value is valid for the rest of the configurations?

If value `wchar_t` is only valid for `Windows`, then add the following:

```
def validate(self):
    if self.options.char_type == "wchar_t" and self.settings.os != Windows:
        raise ConanInvalidConfiguration("Option 'char_type=wchar_t' is only supported in Windows")
```

Reply:
According to https://xerces.apache.org/xerces-c/build-3.html  `wchar_t` is only available on `Windows`.

Using a validate is a much better solution than simply deleting the option, I'll change that.